src/block/request.rs

```diff
@@ -22,20 +26,38 @@
 //! [this](https://docs.oasis-open.org/virtio/virtio/v1.1/csprd01/virtio-v1.1-csprd01.html#x1-280004)).
 //! However, we expect the `Request` interface won't change even if we switch to the general
 //! approach.
+//! - [`RequestExecutor`](struct.RequestExecutor.html) which handles the execution of the block
+//! device request via [`RequestExecutor::execute`](struct.RequestExecutor.html#method.execute)
+//! method. The `RequestExecutor` is wrapping the block device file and keeps, for now, the number
+//! of sectors of this file too. TBD if we keep here the number of sectors or/and the disk size and
+//! other things.
 
-use std::{mem, result};
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::ops::{Deref, DerefMut};
+use std::{io, mem, result};
 
 use crate::queue::DescriptorChain;
 use vm_memory::{
     ByteValued, Bytes, GuestAddress, GuestAddressSpace, GuestMemory, GuestMemoryError,
 };
+use vmm_sys_util::file_traits::FileSync;
+use vmm_sys_util::write_zeroes::{PunchHole, WriteZeroes};
 
 const VIRTIO_BLK_T_IN: u32 = 0;
 const VIRTIO_BLK_T_OUT: u32 = 1;
 const VIRTIO_BLK_T_FLUSH: u32 = 4;
 const VIRTIO_BLK_T_DISCARD: u32 = 11;
 const VIRTIO_BLK_T_WRITE_ZEROES: u32 = 13;
 
+const SECTOR_SHIFT: u8 = 9;
+const SECTOR_SIZE: u64 = (0x01 as u64) << SECTOR_SHIFT;
+
+/// Trait that keeps as supertraits the ones that are necessary for virtio block
+/// request execution on a block device (v1.1 virtio specification).
+pub trait DiskFile: Read + Write + Seek + FileSync + PunchHole + WriteZeroes {}
```

Comment:
I think a more generic name like `Backend` is more suitable, but that's def subject to opinion.